src/basic/fd-util.h

```diff
@@ -126,3 +129,15 @@ static inline char *format_proc_fd_path(char buf[static PROC_FD_PATH_MAX], int f
 
 #define FORMAT_PROC_FD_PATH(fd) \
         format_proc_fd_path((char[PROC_FD_PATH_MAX]) {}, (fd))
+
+/* Determining a file descriptor's path via /proc is not reliable. Only use this function to provide better
+ * log messages when doing fd based path handling! */
+static inline const char *FORMAT_FD_PATH(int fd, char **buf) {
+        assert(buf);
+
+        if (fd == AT_FDCWD)
+                return "";
```

Comment:
shouldn't this return `"."`? i mean, it's called "cwd", as "in current working directory", and that's `.`

Reply:
`AT_FDCWD` can mean either the current working directory or "/", depending on whether the path is absolute or not, so we should return the empty string here.